programmers/모의고사.js

```diff
@@ -0,0 +1,19 @@
+function solution(answers) {
+    const p1 = [1, 2, 3, 4, 5,];
+    const p2 = [2, 1, 2, 3, 2, 4, 2, 5];
+    const p3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5];
+    let answer = [];
+    
+    let s1 = answers.filter( (a, i) => a === p1[i%5] ).length; 
+    let s2 = answers.filter( (a, i) => a === p2[i%8] ).length; 
+    let s3 = answers.filter( (a, i) => a === p3[i%10] ).length; 
+    
+    let max = Math.max(s1, s2, s3);
```

Comment:
1. 변수명 좀 더 명확하게 하기
2. 공백이 이상하게 되있음 신경써줘야 됨
3. 변수 타입 여기선 싹다 const로 해줘야 됨
4. 5, 8, 10 이렇게 보다는 pattern1.length 이런식이 더 나아보임
5. filter메서드를 이용한 풀이는 굿 :)
```suggestion
function solution(answers) {
    const pattern1 = [1, 2, 3, 4, 5,];
    const pattern2 = [2, 1, 2, 3, 2, 4, 2, 5];
    const pattern3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5];
    
    const correct1 = answers.filter((el, i) => el === pattern1[i % pattern1.length]).length; 
    const correct2 = answers.filter((el, i) => el === pattern2[i % pattern2.length]).length; 
    const correct3 = answers.filter((el, i) => el === pattern3[i % pattern3.length]).length; 
    
    const max = Math.max(correct1 , correct2 , correct3 );
    const answer = [];
```